plugins/kubernetes-backend/src/service-locator/SingleTenantServiceLocator.test.ts

```diff
@@ -0,0 +1,160 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '@backstage/backend-common';
+import { Entity } from '@backstage/catalog-model';
+import { ServiceLocatorRequestContext } from '../types/types';
+import { SingleTenantServiceLocator } from './SingleTenantServiceLocator';
+
+describe('SingleTenantConfigClusterLocator', () => {
+  it('empty clusters returns empty cluster details', async () => {
+    const sut = new SingleTenantServiceLocator({
+      getClusters: async () => [],
+    });
+
+    const result = await sut.getClustersByEntity(
+      {} as Entity,
+      {} as ServiceLocatorRequestContext,
+    );
+
+    expect(result).toStrictEqual({ clusters: [] });
+  });
+
+  it('one cluster return from two clusters', async () => {
+    const sut = new SingleTenantServiceLocator({
+      getClusters: async () => {
+        return [
+          {
+            name: 'cluster1',
+            url: 'http://localhost:8080',
+            authMetadata: {},
+          },
+          {
+            name: 'cluster2',
+            url: 'http://localhost:8081',
+            authMetadata: {},
+          },
+        ];
+      },
+    });
+
+    const testEntity = {
+      apiVersion: 'backstage.io/v1alpha1',
+      kind: 'Component',
+      metadata: {
+        namespace: 'default',
+        name: 'testEntity',
+        annotations: {
+          'backstage.io/kubernetes-cluster': 'cluster1',
+        },
+      },
+    };
+
+    const result = await sut.getClustersByEntity(
+      testEntity as Entity,
+      {} as ServiceLocatorRequestContext,
+    );
+
+    expect(result).toStrictEqual({
+      clusters: [
+        {
+          name: 'cluster1',
+          url: 'http://localhost:8080',
+          authMetadata: {},
+        },
+      ],
+    });
+  });
+
+  it('no annotation return all cluster', async () => {
+    const sut = new SingleTenantServiceLocator({
+      getClusters: async () => {
+        return [
+          {
+            name: 'cluster1',
+            url: 'http://localhost:8080',
+            authMetadata: {},
+          },
+          {
+            name: 'cluster2',
+            url: 'http://localhost:8081',
+            authMetadata: {},
+          },
+        ];
+      },
+    });
+
+    const result = await sut.getClustersByEntity(
+      {} as Entity,
+      {} as ServiceLocatorRequestContext,
+    );
+
+    expect(result).toStrictEqual({
+      clusters: [
+        {
+          name: 'cluster1',
+          url: 'http://localhost:8080',
+          authMetadata: {},
+        },
+        {
+          name: 'cluster2',
+          url: 'http://localhost:8081',
+          authMetadata: {},
+        },
+      ],
```

Comment:
Suggestion (non-blocking): Maybe extract this array to a variable and reuse it in the `getClusters` and `expect` statement.